hana/lib/HANAService.js

```diff
@@ -636,19 +636,22 @@ class HANAService extends SQLService {
     }
 
     where(xpr) {
-      return this.xpr({ xpr }, ' = TRUE')
+      return this.xpr({ xpr: [...xpr, 'THEN'] }).slice(0, -4)
     }
 
     having(xpr) {
-      return this.xpr({ xpr }, ' = TRUE')
+      return this.where(xpr)
     }
 
-    xpr({ xpr, _internal }, caseSuffix = '') {
+    xpr(_xpr, caseSuffix = '') {
+      const { xpr, _internal } = _xpr
       // Maps the compare operators to what to return when both sides are null
       const compareOperators = {
-        '=': true,
+        '==': true,
```

Comment:
This would mean to change what we decided and specified?

Reply:
this change makes it as followed:

| xpr | sql |
| --- | --- |
| A = NULL | A IS NULL |
| A = B | A = B |
| A == NULL | A IS NULL |
| A == B | (A = B \|\| A IS NULL AND B IS NULL) |

As part of the reported issue it was determined that `A = NULL` was not being properly converted to `A IS NULL`. That is now resolved by calling `super.operator`. Which will return `IS` or `IS NOT` when applicable. Just like it is done for `SQLite`.

Hope that makes this a bit more clear.